inference-engine/src/inference_engine/cpp_interfaces/interface/ie_iplugin_internal.cpp

```diff
@@ -189,6 +192,10 @@ ICore* IInferencePlugin::GetCore() const noexcept {
     return _core;
 }
 
+const std::shared_ptr<ExecutorManager> & IInferencePlugin::executorManager() const {
```

Comment:
Do we need a special method and special member if we always can get executor manager singleton using `InferenceEngine::executorManager()`?

Reply:
Yes it is required because global instance of executorManager might be destructed. For instance, the CPU plugin uses the executorManager from the destructor. Therefore, IInferencePlugin::executorManager is a safe way for plugins to use it from destructors.